pkg/controller/kccmanager/kccmanager.go

```diff
@@ -113,6 +116,39 @@ func New(ctx context.Context, restConfig *rest.Config, cfg Config) (manager.Mana
 		return nil, fmt.Errorf("error adding schemes: %w", err)
 	}
 
+	// Create a temporary client to read the ConfigConnector object for leader election config.
```

Comment:
Hmmm, this is clever but I just don't know.  We should at least split it out into its own function to avoid accidentally reusing the client.

I think because we aren't going to watch this object, we should rely on the operator to manage the restarts / rollout.

Maybe invent something like a JDBC URL for leader election syntax to make it a one liner, e.g.

`gs://<bucket>/<path>?timeout=15s&namespace=foo`

I'm also not sure who should create the lease objects, should we simply accept a pointer to a pre-configured lease object?